src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -373,6 +392,131 @@ Array<NumericalSolver::Solution> NumericalSolver::integrateDuration(
     return integrateTime(anInitialStateVector, 0.0, aDurationArray, aSystemOfEquations);
 }
 
+NumericalSolver::Solution NumericalSolver::integrateDuration(
+    const StateVector& anInitialStateVector,
+    const Real& aDurationInSeconds,
+    const SystemOfEquationsWrapper& aSystemOfEquations,
+    const Shared<EventCondition>& anEventCondition
+)
+{
+    if (aDurationInSeconds.isZero())
+    {
+        return {anInitialStateVector, 0.0};
```

Comment:
```suggestion
        return {anInitialStateVector, 0.0};
```

Should we use `anInitialStateVector` corresponding instant? for `solution.second` here? (see comment above)